repose-aggregator/components/filters/saml-policy-translation-filter/src/test/scala/org/openrepose/filters/samlpolicy/SamlIdentityClientTest.scala

```diff
@@ -0,0 +1,939 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.samlpolicy
+
+import java.io.ByteArrayInputStream
+import java.nio.charset.{Charset, StandardCharsets}
+import javax.servlet.http.HttpServletResponse._
+import javax.ws.rs.core.MediaType
+
+import org.apache.http.message.BasicHeader
+import org.hamcrest.{Matchers => HM}
+import org.junit.runner.RunWith
+import org.mockito.Mockito.{never, verify, when}
+import org.mockito.{Matchers => MM}
+import org.openrepose.commons.utils.http.CommonHttpHeader.{CONTENT_TYPE, TRACE_GUID}
+import org.openrepose.commons.utils.http.{CommonHttpHeader, ServiceClientResponse}
+import org.openrepose.core.services.serviceclient.akka.{AkkaServiceClient, AkkaServiceClientFactory}
+import org.openrepose.filters.samlpolicy.SamlIdentityClient.{OverLimitException, SC_TOO_MANY_REQUESTS}
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FunSpec, Matchers}
+import play.api.libs.json.Json
+
+import scala.language.implicitConversions
+import scala.util.{Failure, Success}
+
+@RunWith(classOf[JUnitRunner])
+class SamlIdentityClientTest extends FunSpec with BeforeAndAfterEach with Matchers with MockitoSugar {
+
+  var akkaServiceClient: AkkaServiceClient = _
+  var akkaServiceClientFactory: AkkaServiceClientFactory = _
+  var samlPolicyProvider: SamlIdentityClient = _
+
+  override def beforeEach(): Unit = {
+    akkaServiceClient = mock[AkkaServiceClient]
+    akkaServiceClientFactory = mock[AkkaServiceClientFactory]
+
+    samlPolicyProvider = new SamlIdentityClient(akkaServiceClientFactory)
+
+    when(akkaServiceClientFactory.newAkkaServiceClient())
+      .thenReturn(akkaServiceClient)
+    when(akkaServiceClientFactory.newAkkaServiceClient(MM.anyString()))
```

Comment:
I'm a terrible person. I do this same aliasing, but it just occurred to me that we could instead alias `anyString` to `neString`. Please don't do that, because it's terrible. I feel like its the code equivalent of a pun.